source/common/http/matching/inputs.cc

```diff
@@ -2,13 +2,47 @@
 
 #include "envoy/registry/registry.h"
 
+#include "source/common/http/utility.h"
+
 namespace Envoy {
 namespace Http {
 namespace Matching {
+
+Matcher::DataInputGetResult
```

Comment:
nit: The implementation here is thicker so that it is suitable to be put in .cc file even though get() function for `class HttpHeadersDataInputBase` sits in .h file.

I don't have strong preference between .cc and .h file. But can we be consistent on this ---- Either move `HttpHeadersDataInputBase`'s get to .cc OR move it to .h.
WDYT?

Reply:
I just saw that `status_code_input` has its separate CC/header and proto files. Maybe I should do the same here?

Reply:
Right in this exact file, [class HttpHeadersDataInputBase](https://github.com/envoyproxy/envoy/blob/34d1e7ddaaffe31b907f13502d96c7a36dacd67b/source/common/http/matching/inputs.h#L25) has get()'s both impl and declare in the header file.

I think `HttpHeadersDataInputBase` is more related here and we should keep consist this input.{cc.h} at least.

Reply:
So in the end I should put the code of the `get` function in `inputs.h`?

Reply:
Yea, that works for me.